src/app/systelab-components/datepicker/datepicker.component.ts

```diff
@@ -121,6 +132,33 @@ export class Datepicker implements OnInit, AfterViewInit, DoCheck, OnDestroy {
 		}
 	}
 
+	private checkTooFarDate() {
+
+		if (this._currentDate) {
+			this._currentDate.setHours(0, 0, 0, 0);
+			const futureDate = this.addDaysToDate(new Date(), this.maxDaysAfter, false);
+
+			if (this._currentDate.getTime() >= futureDate.getTime()) {
+				this.tooFarDate = true;
+			} else {
+				this.tooFarDate = false;
+			}
+
+		} else {
+			this.tooFarDate = false;
+		}
+	}
+
+	private addDaysToDate(referenceDate: Date, daysToAdd: number, setTimeToMidnight: boolean): Date {
```

Comment:
Use existing library date-fns to add days

Reply:
Add documentation to README.md. Specify markPreviousAfterDate is (deprecated) indicating that now the new inputs should be used